packages/local_auth/local_auth_android/android/src/main/java/io/flutter/plugins/localauth/LocalAuthPlugin.java

```diff
@@ -274,6 +274,7 @@ private boolean canAuthenticateWithDeviceCredential() {
     if (Build.VERSION.SDK_INT < 30) {
       // Checking/setting device credential only authentication is not allowed before API 11,
       // so check for presence of PIN, pattern, or password instead.
+      if (keyguardManager == null) return false;
       return keyguardManager.isDeviceSecure();
```

Comment:
@camsim99 I think u need to update the code to
```
iif (Build.VERSION.SDK_INT >= 16 30 && Build.VERSION.SDK_INT < 30) {
```

`keyguardManager.isDeviceSecure()` only works on M and above.

<img width="924" alt="Screenshot 2022-10-07 at 5 15 39 PM" src="https://user-images.githubusercontent.com/42466125/194519096-92e264f0-c317-4c63-9207-d022f69770e9.png">

or you can just use what i wrote before

```
private boolean canAuthenticateWithDeviceCredential() {
    //supports from Android 23 onwards
    if ( Build.VERSION.SDK_INT >= Build.VERSION_CODES.R ) {
      if (biometricManager == null) return false;
      return biometricManager.canAuthenticate(BiometricManager.Authenticators.DEVICE_CREDENTIAL)
          == BiometricManager.BIOMETRIC_SUCCESS;
    }else if ( Build.VERSION.SDK_INT >= Build.VERSION_CODES.M ) {
      if (keyguardManager == null) return false;
      return keyguardManager.isDeviceSecure();
    }
    return false;
}
```

Reply:
We actually don't authenticate at all below API 23 (M) in order to be compatible with the biometrics library:
https://github.com/flutter/plugins/blob/5a0209d99d8966e0907f9e0632558234f8dca71c/packages/local_auth/local_auth_android/android/src/main/java/io/flutter/plugins/localauth/LocalAuthPlugin.java#L142-L146

You are right about needing an explicit check for that, so I refactored the code a bit.